Renaming the WASI target in Rust 1.71.0 (#1099)
rust-lang/blog.rust-lang.org: posts/2023-05-01-wasi-target-rename-1.71.0.md

reintroduce it in the future as the target name for an eventual 1.0 stable WASI
release.

While we don’t have any immediate plans to deprecate the Preview 1 target, the
reality is that most of the development of WASI in recent years has centered
around WASI Preview 2 and [Wasm Components][components]. The expectation is that
Preview 2 will eventually supplant Preview 1, and support across the WASI
ecosystem will gradually wind down for Preview 1. By adopting “preview” in the
target names, we can begin to more clearly communicate that the current WASI
targets should be considered development snapshots, and will eventually be
superseded.

Work on adding a WASI Preview 2 target to the compiler is currently underway,
but is not yet complete. We expect this to land in a future release of Rust. The
target triple for this target will be `wasm32-wasi-preview2`.

## Migrating Targets

To upgrade to the new WASI target on Rust 1.71 nightly you can run the
following commands using [rustup](https://rustup.rs):

```bash
$ rustup +nightly target remove wasm32-wasi
$ rustup upgrade nightly
$ rustup +nightly target add wasm32-wasi-preview1

> ```

Reviewer: Perhaps we should also mention that `cargo build --target wasm32-wasi` is now `cargo build --target wasm32-wasi-preview1`?

---
layout: post
title: "Renaming the WASI target in Rust 1.71.0"
author: Yoshua Wuyts
---
In Rust 1.71 the existing `wasm32-wasi` target will be renamed

> to `wasm32-wasi-preview1`. This matches the naming introduced in the [WASI

Reviewer 2: ```suggestion
to `wasm32-wasip1`. This matches the naming introduced in the [WASI
```

Preview Framework][wpf], whose scheme has already been adopted by other
language toolchains such as [Golang] and [Node.js]. It is expected that
eventually WASI will eventually stabilize far enough that it will reach a 1.0
status. By freeing up the `wasm32-wasi` target name now, it enables us to

> reintroduce it in the future as the target name for an eventual 1.0 stable WASI

Reviewer 2: I hesitate to use the term stable here since these targets are also stable they just hint that they will be superseded by a future release. Perhaps instead of call the eventual 1.0 release "stable" we can say something akin to:

> By freeing up the `wasm32-wasi` target name now, it enables us to
reintroduce it in the future as the target name for an eventual 1.0 WASI release where evolution of WASI is expected to slow considerably in pace. All wasi targets are in and of themselves stable, but it's important to note that future wasi releases are expected to supersede previous versions which will not receive feature updates.

release.

While we don’t have any immediate plans to deprecate the Preview 1 target, the
reality is that most of the development of WASI in recent years has centered
around WASI Preview 2 and [Wasm Components][components]. The expectation is that
Preview 2 will eventually supplant Preview 1, and support across the WASI

> ecosystem will gradually wind down for Preview 1. By adopting “preview” in the

Reviewer 2: > By adopting “preview” in the target names

Note: "preview" isn't actually used in the target names just "p" which stands for preview.

target names, we can begin to more clearly communicate that the current WASI
targets should be considered development snapshots, and will eventually be
superseded.

Work on adding a WASI Preview 2 target to the compiler is currently underway,
but is not yet complete. We expect this to land in a future release of Rust. The

> target triple for this target will be `wasm32-wasi-preview2`.

Reviewer 2: ```suggestion
target triple for this target will be `wasm32-wasip2`.
```

## Migrating Targets

To upgrade to the new WASI target on Rust 1.71 nightly you can run the
following commands using [rustup](https://rustup.rs):

```bash
$ rustup +nightly target remove wasm32-wasi
$ rustup upgrade nightly

> $ rustup +nightly target add wasm32-wasi-preview1
> $ cargo build --target wasm32-wasi-preview1

Reviewer 2: ```suggestion
$ rustup +nightly target add wasm32-wasip1
$ cargo build --target wasm32-wasip1
```
